assets/js/components/higherorder/withdata.js

```diff
@@ -62,7 +62,12 @@ import { __ } from '@wordpress/i18n';
  *                                                 if the data is "zero".
  * @param {function}        getDataError           A callback function that is passed the resolved data and returns the
  *                                                 error message.
-
+ *
+ * @return {React.Component} Component  	Returns React.Components based on data and state.
+ * 											If has data  	  Return DataDependentComponent with data.
```

Comment:
This is sort of a weird format, but I dunno how else to handle it. I think JSDoc is kinda meh, so I guess this is fine 😅